service/src/main/java/bio/terra/workspace/db/WorkspaceDao.java

```diff
@@ -222,145 +230,131 @@ private boolean updateWorkspaceColumns(UUID workspaceId, MapSqlParameterSource c
+   * @param cloudPlatform cloud platform enum
+   * @param context serialized cloud context attributes
+   * @param flightId calling flight id. Used to ensure that we do not delete a good context while
+   *     undoing from trying to create a conflicting context.
+   */
+  private void createCloudContextWorker(
+      UUID workspaceId, CloudPlatform cloudPlatform, String context, String flightId) {
+    final String platform = cloudPlatform.toSql();
     final String sql =
-        "INSERT INTO cloud_context (workspace_id, cloud_platform, context)"
-            + " VALUES (:workspace_id, :cloud_platform, :context::json)";
+        "INSERT INTO cloud_context (workspace_id, cloud_platform, context, creating_flight)"
+            + " VALUES (:workspace_id, :cloud_platform, :context::json, :creating_flight)";
     MapSqlParameterSource params =
         new MapSqlParameterSource()
             .addValue("workspace_id", workspaceId.toString())
-            .addValue("cloud_platform", CloudPlatform.GCP.toString())
-            .addValue("context", serializeGcpCloudContext(cloudContext));
+            .addValue("cloud_platform", platform)
+            .addValue("context", context)
+            .addValue("creating_flight", flightId);
     try {
       jdbcTemplate.update(sql, params);
-      logger.info("Inserted record for GCP cloud context for workspace {}", workspaceId);
+      logger.info("Inserted record for {} cloud context for workspace {}", platform, workspaceId);
     } catch (DuplicateKeyException e) {
       throw new DuplicateCloudContextException(
-          String.format("Workspace with id %s already has context for GCP cloud type", workspaceId),
+          String.format(
+              "Workspace with id %s already has context for %s cloud type", workspaceId, platform),
           e);
     }
   }
 
   /**
-   * Delete the GCP cloud context for a workspace
+   * Common worker for retrieving cloud context
    *
-   * @param workspaceId workspace of the cloud context
+   * @param workspaceId workspace of the context
+   * @param cloudPlatform platform context to retrieve
+   * @return empty or the serialized cloud context info
    */
-  @WriteTransaction
-  public void deleteGcpCloudContext(UUID workspaceId) {
-    deleteGcpCloudContextWorker(workspaceId);
+  private Optional<String> getCloudContextWorker(UUID workspaceId, CloudPlatform cloudPlatform) {
+    String sql =
+        "SELECT context FROM cloud_context "
+            + "WHERE workspace_id = :workspace_id AND cloud_platform = :cloud_platform";
+    MapSqlParameterSource params =
+        new MapSqlParameterSource()
+            .addValue("workspace_id", workspaceId.toString())
+            .addValue("cloud_platform", cloudPlatform.toSql());
+    try {
+      return Optional.ofNullable(
+          DataAccessUtils.singleResult(
+              jdbcTemplate.query(sql, params, (rs, rowNum) -> rs.getString("context"))));
+    } catch (EmptyResultDataAccessException e) {
```

Comment:
I think `singleResult` returns `null` instead of throwing `EmptyResultDataAccessException`. https://docs.spring.io/spring-framework/docs/current/javadoc-api/org/springframework/dao/support/DataAccessUtils.html#singleResult-java.util.Collection-

Reply:
Yes, you are right. Don't know why the try-catch was there. I'll remove it.